src/applications/model/gaming-streaming-server.cc

```diff
@@ -0,0 +1,235 @@
+        }
+      else if (Ipv6Address::IsMatchingType(m_peerAddress) == true)
+        {
+          if (m_socket->Bind6 () == -1)
+            {
+              NS_FATAL_ERROR ("Failed to bind socket");
+            }
+          m_socket->Connect (Inet6SocketAddress (Ipv6Address::ConvertFrom (m_peerAddress), m_peerPort));
+        }
+      else if (InetSocketAddress::IsMatchingType (m_peerAddress) == true)
+        {
+          if (m_socket->Bind () == -1)
+            {
+              NS_FATAL_ERROR ("Failed to bind socket");
+            }
+          m_socket->Connect (m_peerAddress);
+        }
+      else if (Inet6SocketAddress::IsMatchingType (m_peerAddress) == true)
+        {
+          if (m_socket->Bind6 () == -1)
+            {
+              NS_FATAL_ERROR ("Failed to bind socket");
+            }
+          m_socket->Connect (m_peerAddress);
+        }
+      else
+        {
+          NS_ASSERT_MSG (false, "Incompatible address type: " << m_peerAddress);
+        }
+    }
+  m_socket->SetRecvCallback (MakeNullCallback<void, Ptr<Socket> > ());
+  m_socket->SetAllowBroadcast (true);
+  m_sendEvent = Simulator::Schedule (Seconds (0.0), &GamingStreamingServer::Send, this);
+
+}
+
+void
+GamingStreamingServer::StopApplication ()
+{
+  NS_LOG_FUNCTION (this);
+  Simulator::Cancel (m_sendEvent);
+}
+
+void
+GamingStreamingServer::SendPacket (Ptr<RandomVariableStream> pktDist, Ptr<RandomVariableStream> iatDist)
+{
+  uint32_t size = pktDist->GetInteger ();
+  NS_LOG_FUNCTION (this << size);
+  Ptr<Packet> p;
+  uint32_t packetSize;
+  if (size > 12)
+  {
+    packetSize = size - 12; // 12 is the size of the SeqTsHeader
+  }
+  else
+  {
+    packetSize = 0;
+  }
+  p = Create<Packet> (packetSize);
+  SeqTsHeader seqTs;
+  seqTs.SetSeq (m_sentSeq);
+  p->AddHeader (seqTs);
+  TimestampTag timestamp;
+  timestamp.SetTimestamp (Simulator::Now ());
+  p->AddByteTag (timestamp);
+  std::stringstream addressString;
+  if (Ipv4Address::IsMatchingType (m_peerAddress) == true)
+    {
+      addressString << Ipv4Address::ConvertFrom (m_peerAddress);
+    }
+  else if (Ipv6Address::IsMatchingType (m_peerAddress) == true)
+    {
+      addressString << Ipv6Address::ConvertFrom (m_peerAddress);
+    }
+  else
+    {
+      addressString << m_peerAddress;
+    }
+
+  if ((m_socket->Send (p)) >= 0)
+    {
+      m_totSentPackets++;
+      m_sentSeq++;
+      NS_LOG_INFO ("Sent " << size << " bytes to "
+                           << addressString.str ());
+    }
+  else
+    {
+      NS_LOG_INFO ("Error while sending " << size << " bytes to "
+                                          << addressString.str ());
+    }
+
+  m_sendEvent = Simulator::Schedule (MilliSeconds (iatDist->GetInteger ()),
+                                     &GamingStreamingServer::SendPacket, this, pktDist, iatDist);
+}
+
+void
+GamingStreamingServer::Send (void)
+{
+  NS_LOG_FUNCTION (this);
+  SendPacket (m_pks, m_iat);
+}
```

Comment:
In this case, I need to overload the StartApplication for all derived classes, which cause a huge code duplication.
Now, I just overload a `Send` method.
Another approach could be,  instead of saving a single RV for `m_packetSize` and `m_interArrivalTime`, save a vector of RVs and in the StartApplication, iterate over them and call SendPacket for different RVs.

Reply:
the next comments basically override this one, i forgot to remove it. Please take a look at the whole review to fully understand what has been suggested